### PR TITLE
Create battle request

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: database/schemas/BattleSchema.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+      extraLife: { type: Number },
+    },
+    secondPlayer: {
+      cryptoName: { type: String },
+      playerID: { type: String },
+      extraLife: { type: Number },
+    },
+    healthPoints: { type: Number, required: true },
+  },
+  steps: { type: Array },
```
Comment (crypto-wars-wd) [inline database/schemas/BattleSchema.js]: add default empty array

File: test/controllers/battleController.test.js [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    let data, result;
+    beforeEach(async () => {
+      data = {
+        cryptoName: `${faker.name.firstName()}`,
+        playerID: `${faker.name.firstName()}${faker.random.number()}`,
+        healthPoints: faker.random.number(),
+      };
+      result = await chai.request(app)
+        .post('/api/create-battle')
+        .send(data);
```
Comment (crypto-wars-wd) [inline test/controllers/battleController.test.js:16]: find battle in DB and check, that it was cteated, it is can be just 1 test

File: test/factories/BattleFactory.js [hunk 3]
```diff
@@ -0,0 +1,18 @@
+const { faker, models } = require('test/testHelper');
```
Comment (crypto-wars-wd) [inline test/factories/BattleFactory.js]: why you require all models? You have Battle at testHelper exports

File: test/factories/BattleFactory.js [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+  const battleData = {
+    'playersInfo.firstPlayer.cryptoName': cryptoName || `${faker.name.firstName()}`,
+    'playersInfo.firstPlayer.playerID': playerID || `${faker.name.firstName()}${faker.random.number()}`,
+    'playersInfo.healthPoints': healthPoints || `${faker.random.number()}`,
+  };
+
+  const battle = new models.Battle(battleData);
+
+  await battle.save();
+  return battle;
```
Comment (crypto-wars-wd) [inline test/factories/BattleFactory.js:16]: toObject(), and better return it like object {battle}

File: test/models/battleModel.test.js [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+    });
+    it('should find battle', async () => {
+      await battleModel.createNewBattle(battleData);
+      const battle = await models.Battle.findOne({ 'playersInfo.firstPlayer.cryptoName': 'BTC' });
+      expect(battle.playersInfo.firstPlayer.cryptoName).to.be.eq('BTC');
+    });
+    it('should be error message', async () => {
+      const { message } = await battleModel.createNewBattle({ lkkdfs: 'dfsdf' });
+      expect(message).to.be.exist;
+    });
```
Comment (crypto-wars-wd) [inline test/models/battleModel.test.js]: I don’t quite understand what you are doing if you want to test the creation of a battle. You don’t need to create it in the factory, look at the parser how we use the factory to return only data, without creating a record, and do the same

**Review by ghost — COMMENTED** [review]
Fix it a little bit

File: database/schemas/BattleSchema.js (hunk 1 above)
Comment (crypto-wars-wd) [reply database/schemas/BattleSchema.js:19]: done

Comment (Fesmofet) [reply database/schemas/BattleSchema.js]: done

File: test/factories/BattleFactory.js (hunk 3 above)
Comment (Fesmofet) [reply test/factories/BattleFactory.js]: done

File: test/factories/BattleFactory.js (hunk 4 above)
Comment (Fesmofet) [reply test/factories/BattleFactory.js:16]: done

File: test/models/battleModel.test.js (hunk 5 above)
Comment (Fesmofet) [reply test/models/battleModel.test.js]: done

File: test/controllers/battleController.test.js (hunk 2 above)
Comment (Fesmofet) [reply test/controllers/battleController.test.js:16]: done

**Review by olegtsigulov — APPROVED** [review]

**Pull request merged**